src/Entity/Plan.php

```diff
@@ -0,0 +1,343 @@
+
+/**
+ * Wrapper around a Stripe plan
+ *
+ * This entity encapsulates most Stripe attributes, but adds its own set of properties. For instance, it
+ * supports a "features" array that you can display in your pricing pages, as well as a searchable "limits"
+ * collection that you can use it if you need to attribute a specific billable resource to a plan according
+ * to some pre-defined limits (like a maximum number of private repositories)
+ *
+ * @author  Michaël Gallego <mic.gallego@gmail.com>
+ * @licence MIT
+ */
+class Plan
+{
+    const INTERVAL_DAY   = 'day';
+    const INTERVAL_WEEK  = 'week';
+    const INTERVAL_MONTH = 'month';
+    const INTERVAL_YEAR  = 'year';
+
+    /**
+     * @var int
+     */
+    protected $id;
+
+    /**
+     * @var string
+     */
+    protected $stripeId;
+
+    /**
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * @var int
+     */
+    protected $amount;
+
+    /**
+     * @var string
+     */
+    protected $currency;
+
+    /**
+     * @var string
+     */
+    protected $interval;
+
+    /**
+     * @var int
+     */
+    protected $intervalCount;
+
+    /**
+     * @var int|null
+     */
+    protected $trialPeriodDays;
+
+    /**
+     * @var DateTime
+     */
+    protected $createdAt;
+
+    /**
+     * @var array
+     */
+    protected $features = [];
+
+    /**
+     * @var array|\Doctrine\Common\Collections\Collection
+     */
+    protected $limits = [];
+
+    /**
+     * @var bool
+     */
+    protected $active = false;
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        $this->limits = new ArrayCollection();
+    }
+
+    /**
+     * @return int
+     */
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param string $stripeId
+     */
+    public function setStripeId($stripeId)
+    {
+        $this->stripeId = (string) $stripeId;
+    }
+
+    /**
+     * @return string
+     */
+    public function getStripeId()
+    {
+        return $this->stripeId;
+    }
+
+    /**
+     * @param string $name
+     */
+    public function setName($name)
+    {
+        $this->name = (string) $name;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * @param int $amount
+     */
+    public function setAmount($amount)
+    {
+        $this->amount = (int) $amount;
+    }
+
+    /**
+     * @return int
+     */
+    public function getAmount()
+    {
+        return $this->amount;
+    }
+
+    /**
+     * @param string $currency
+     */
+    public function setCurrency($currency)
+    {
+        $this->currency = (string) $currency;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCurrency()
+    {
+        return $this->currency;
+    }
+
+    /**
+     * @param  string $interval
+     * @throws InvalidArgumentException
+     */
+    public function setInterval($interval)
```

Comment:
That's Stripe naming. I have a one-to-one mapping. For instance a plan that bills each month, interval == month, intervalCount = 1